lisa/sut_orchestrator/baremetal/cluster/idrac.py

```diff
@@ -80,26 +108,81 @@ def type_schema(cls) -> Type[schema.TypedSchema]:
     def get_start_stop(self) -> Type[features.StartStop]:
         return IdracStartStop
 
+    def get_serial_console(self) -> Type[features.SerialConsole]:
+        return IdracSerialConsole
+
     def deploy(self, environment: Environment) -> Any:
         self.login()
         self._eject_virtual_media()
-        self._change_boot_order_once("VCD-DVD")
         assert self.client.iso_http_url, "iso_http_url is required for idrac client"
-        if self.get_power_state() == "Off":
-            self._log.debug("System is already off.")
-        else:
-            self.reset("GracefulShutdown")
+        self._change_boot_order_once("VCD-DVD")
+        self.reset("ForceOff")
         self._insert_virtual_media(self.client.iso_http_url)
-        self.reset("On")
+        self.reset("On", force_run=True)
+        self.logout()
+
+    def cleanup(self) -> None:
+        self.login()
+        self._clear_serial_console_log()
         self.logout()
 
-    def reset(self, operation: str) -> None:
+    def get_client_capabilities(self, client: ClientSchema) -> ClientCapabilities:
+        if client.capabilities:
+            return client.capabilities
+        self.login()
+        response = self.redfish_instance.get(
+            "/redfish/v1/Systems/System.Embedded.1/",
+        )
+        cluster_capabilities = ClientCapabilities()
+        cluster_capabilities.core_count = int(
+            response.dict["ProcessorSummary"]["LogicalProcessorCount"]
+        )
+        cluster_capabilities.free_memory_mb = (
+            int(response.dict["MemorySummary"]["TotalSystemMemoryGiB"]) * 1024
+        )
+        self.logout()
+        return cluster_capabilities
+
+    def get_serial_console_log(self) -> str:
+        response = self.redfish_instance.post(
+            "/redfish/v1/Managers/iDRAC.Embedded.1/SerialInterfaces"
+            "/Serial.1/Actions/Oem/DellSerialInterface.SerialDataExport",
+            body={},
+        )
+        check_till_timeout(
+            lambda: int(response.status) == 200,
+            timeout_message="wait for response status 200",
+        )
+        return str(response.text)
+
+    def get_server_screen_shot(self, file_type: str = "ServerScreenShot") -> str:
+        response = self.redfish_instance.post(
+            "/redfish/v1/Dell/Managers/iDRAC.Embedded.1/DellLCService/Actions/"
+            "DellLCService.ExportServerScreenShot",
+            body={"FileType": file_type},
+        )
+        self._wait_for_completion(response)
+        return str(response.dict["ServerScreenShotFile"])
+
+    def reset(self, operation: str, force_run: bool = False) -> None:
+        if not force_run and operation in self.state_dict.keys():
+            if self.get_power_state() == self.state_dict[operation]:
```

Comment:
Get the dict value one time, and use it in next places. So the logic would be easier to understand, and lines shorter.